code/modules/antagonists/changeling/abilities/regeneration.dm

```diff
@@ -174,24 +174,67 @@
 	lock_holder = FALSE
 	ignore_holder_lock = 1
 
+	incapacitationCheck()
+		return 0
+
 	cast(atom/target)
 		if (..())
 			return 1
-		if (tgui_alert(holder.owner,"Are we sure?","Speed Regenerate?",list("Yes","No")) != "Yes")
-			return 1
 
-		if (!src.cooldowncheck())
-			boutput(holder.owner, "<span class='alert'>That ability is on cooldown for [round((src.last_cast - world.time) / 10)] seconds.</span>")
+		var/datum/abilityHolder/changeling/H = holder
+		if (!istype(H))
+			boutput(holder.owner, "<span class='alert'>That ability is incompatible with our abilities. We should report this to a coder.</span>")
 			return 1
 
 		var/mob/living/carbon/human/C = holder.owner
-		if (!istype(C))
-			boutput(holder.owner, "<span class='alert'>We have no idea what we are, but it's damn sure not compatible.</span>")
+		if (tgui_alert(C,"Are we sure?","Speed regen?",list("Yes","No")) != "Yes")
+			boutput(holder.owner, "<span class='notice'>We change our mind.</span>")
 			return 1
-		boutput(holder.owner, "<span class='notice'>Your skin begins reforming around your skeleton.</span>")
 
-		while(C.health < C.max_health || !C.limbs.l_arm || !C.limbs.r_arm || !C.limbs.l_leg || !C.limbs.r_leg)
-			if(isdead(C))
-				break
-			sleep(3 SECONDS)
-			changeling_super_heal_step(C)
+		C.changeStatus("changeling_speedregen", 30 SECONDS)
+		return 0
+
+// changeling speedregen status effect//
+/datum/statusEffect/c_regeneration
+
+	id = "changeling_speedregen"
+	name = "Speed regeneration"
+	desc = "You quickly heal the damage dealt to you."
+	icon_state = "heart+"
+	unique = TRUE
+	maxDuration = 30 SECONDS
+
+	onAdd(optional=null)
+		. = ..()
+		var/mob/living/carbon/human/C
+		if (ishuman(owner))
+			C = owner
+			boutput(C, "<span class='notice'>We start regenerating.</span>")
+			C.visible_message("<span class='alert'><B>[C]'s flesh starts moving and sliding around oddly, repairing their wounds!</B></span>")
+			return
+		else
+			owner.delStatus("changeling_speedregen")
+
+	onUpdate(timePassed)
+		var/mob/living/carbon/human/C
+		if (ishuman(owner))
+			C = owner
+			if (!C.getStatusDuration("burning"))
+				C.HealDamage("All", 2, 2, 1)
+				if (prob(20))
+					C.visible_message("<span class='alert'><B>[C]'s flesh is moving and sliding around oddly!</B></span>")
+					boutput(C, "<span class='notice'>We feel our flesh knitting back together.</span>")
+					C.HealDamage ("All", 3, 2, 2)
+			else // lings are vulnerable to fire so it stopping their regen made sense to me
+				if(prob(25))
```

Comment:
Maybe a cooldown check instead?

Reply:
can you give a quick example since i cant really get it to work?

Reply:
if (!ON_COOLDOWN(H, "cling_regen_fire_cancellation", 3 SECONDS))

Reply:
did the same thing before that, seems to send out multiple messages at once and then go on cooldown and repeat

Reply:
Hm, you could maybe try a GET_COOLDOWN() check before, otherwise dunno

Reply:
hmmm still cant seem to get it to work, maybe keep it a probablity?
I assume it happens because its inside of onUpdate so it activates a cooldown every time proc is called

Reply:
`ON_COOLDOWN` doesn't refresh the cooldown if it's still cooling down, it should work fine if you just replace the `prob` check

Reply:
it wasnt working because i put if in the wrong place (silly goose moment)